pkg/util/deployment/deployment.go

```diff
@@ -30,22 +30,34 @@ import (
 
 // Returns the old RCs targetted by the given Deployment.
 func GetOldRCs(deployment extensions.Deployment, c client.Interface) ([]*api.ReplicationController, error) {
+	return GetOldRCsCustomized(deployment, c,
+		func(namespace string) (*api.PodList, error) {
+			return c.Pods(namespace).List(labels.SelectorFromSet(deployment.Spec.Selector), fields.Everything())
+		},
+		func(namespace string) ([]api.ReplicationController, error) {
+			rcList, err := c.ReplicationControllers(namespace).List(labels.Everything(), fields.Everything())
```

Comment:
We could also add a generalized label in replication controllers owned by a deployment and filter based on it here.